src/menu/create_wallet_menu.c

```diff
@@ -0,0 +1,145 @@
+
+#include "constant_texts.h"
+#include "ui_menu.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef enum {
+  GENERATE_NEW_WALLET,
+  RESTORE_FROM_SEED,
+} create_wallet_options_e;
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static const evt_config_t create_wallet_evt_config = {
+    .abort_disabled = false,
+    .evt_selection.byte = EVT_CONFIG_UI_EVT,
+    .timeout = MAX_INACTIVITY_TIMEOUT};
+
+const flow_step_t create_wallet_step = {
+    .step_init_cb = create_wallet_menu_initialize,
+    .p0_cb = NULL,
+    .ui_cb = create_wallet_menu_handler,
+    .usb_cb = NULL,
+    .nfc_cb = NULL,
+    .evt_cfg_ptr = &create_wallet_evt_config,
+    .flow_data_ptr = NULL};
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+void create_wallet_menu_initialize(const engine_ctx_t *ctx,
+                                   const void *data_ptr) {
+  /* Create an array of pointers holding the string to display on the menu. */
+  char *menu_option_ptr_array[NUMBER_OF_OPTIONS_NEW_WALLET];
+
+  for (uint8_t menu_index = 0; menu_index < NUMBER_OF_OPTIONS_NEW_WALLET;
+       menu_index++) {
+    menu_option_ptr_array[menu_index] =
+        (char *)ui_text_options_new_wallet[menu_index];
+  }
+
+  menu_init((const char **)menu_option_ptr_array,
+            NUMBER_OF_OPTIONS_NEW_WALLET,
+            ui_text_heading_new_wallet,
+            true);
+
+  return;
+}
+
+void create_wallet_menu_handler(const engine_ctx_t *ctx,
+                                ui_event_t ui_event,
+                                const void *data_ptr) {
+  if (ui_event.event_type == UI_EVENT_REJECT) {
+    engine_delete_current_flow_step(ctx);
+  }
+}
```

Comment:
add a TODO please for rest of the ui events.

Reply:
will take this separately in a different pr